core/validation/specific_validations/towns_fund_round_four.py

```diff
@@ -200,6 +201,36 @@ def validate_sign_off(workbook: dict[str, pd.DataFrame]) -> list["TownsFundRound
         return failures
 
 
+def validate_psi_funding_gap(workbook: dict[str, pd.DataFrame]) -> list["TownsFundRoundFourValidationFailure"] | None:
+    """Validates that if a funding gap of > 0 is specified in Private funding sources tab that an additional comment
+    is supplied to explain why
+
+    :param workbook: A dictionary where keys are sheet names and values are pandas
+                     DataFrames representing each sheet in the Round 4 submission.
+    :return: ValidationErrors
+    """
+    sheet = workbook["Private Investments"]
+    if (
+        len(
+            sheet.loc[
+                (sheet["Private Sector Funding Required"] > sheet["Private Sector Funding Secured"])
+                & (sheet["Additional Comments"].isna())
+            ]
+        )
+        > 0
+    ):
```

Comment:
This might be easier to follow if there isn't so much being done on one line. Something like:
```
psi_df = workbook["Private Investments"]
invalid_mask = (psi_df["Private Sector Funding Required"] > psi_df["Private Sector Funding Secured"]) & (
    psi_df["Additional Comments"].isna()
)
invalid_psi_rows = psi_df.loc[invalid_mask]
if len(invalid_psi_rows) > 0:
```